tests/integration/test_dbapi_integration.py

```diff
@@ -158,6 +471,54 @@ def test_array_query_param(trino_connection):
     assert rows[0][0] == "array(integer)"
 
 
+def test_array_none_query_param(trino_connection):
+    cur = trino_connection.cursor(experimental_python_types=True)
+
+    params = [None, None]
```

Comment:
It'd be useful to have a mix of None and non-None value in the array as well (which I what I originally meant when asking to add this test - but thanks for covering NULL arrays as well - I indeed missed that). What I was trying to test was that whether we correctly infer the type of the array.

Reply:
Python doesn't have typed arrays. the type is determined for every value. I guess if we start mixing types, Trino will error. Combining None and another type should just work. Will add a test for that.

Reply:
None + another type should work in Trino provided we are CASTing the NULLs according to type of other elements in array.

Reply:
If such NULL + non-NULL arrays don't work correctly when translating to the query we should block them within the client maybe.

Reply:
Resolved by adding `test_array_none_and_another_type_query_param`

It does work correctly.